internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -399,6 +471,17 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 			return reconcile.Result{RequeueAfter: shortWait}, nil
 		}
 
+		// DeepCopy composed resource to ensure changes to composed are not persisted
+		// when fetching connection details or readiness
+		cdCopy := cd.DeepCopyObject()
+		for _, p := range comp.Spec.Resources[i].Patches {
+			if err := p.Apply(patchObject, cdCopy); err != nil {
```

Comment:
Patches from composed to composite currently can take up to 60s to apply due to the long wait reconcile. One option if this is not desirable is we could detect if the used composition has any patches which may patch the composite and shorten the longWait reconcile for these resources.